caikit/runtime/model_management/model_manager.py

```diff
@@ -325,18 +326,19 @@ def unload_model(self, model_id) -> int:
             Model_size (int) : Size of the loaded model in bytes
         """
         log.debug("List of loaded models: %s", str(self.loaded_models))
-        # If the model failed to load, just return 0; no need to throw an error here.
-        if model_id not in self.loaded_models:
-            log.debug("Model '%s' is not loaded, so it cannot be unloaded!", model_id)
-            return 0
+        try:
```

Comment:
Rearranged here for a small efficiency to avoid a potential race around looking up in the map twice (once to check presence and once again to remove)